src/Logger/ElasticsearchLogger.php

```diff
@@ -249,40 +271,39 @@ protected function write(array $record)
             $body[EmsFields::LOG_USERNAME_FIELD] = $this->user;
             $body[EmsFields::LOG_IMPERSONATOR_FIELD] = $this->impersonator;
 
-            $this->bulk[] = [
-                'index' => [
-                    '_type' => 'doc',
-                    '_index' => 'ems_internal_logger_index_'.(new DateTime())->format('Ymd'),
-                ],
-            ];
-            $this->bulk[] = $body;
+            $action = new Action();
+            $action->setIndex(self::EMS_INTERNAL_LOGGER_INDEX.(new DateTime())->format('Ymd'));
+
+            if (\version_compare($this->client->getVersion(), '7') < 0) {
+                $action->setType('doc');
+            }
+            $action->setOpType(Action::OP_TYPE_INDEX);
+            $action->setSource($body);
 
-            if (\count($this->bulk) > 200) {
+            $this->bulk->addAction($action);
+            if (\count($this->bulk->getActions()) > 200) {
                 $this->treatBulk();
```

Comment:
where is the treatBulk method? Should it be $this->bulk->treatBulk();

Reply:
The treatBulk is just bellow, it handles the fact that we may still receive logs after the onKernelTerminate event (logs that we don't want/able to save i.e. from the Symfony the debug bar)